### ISSUE TITLE
Serious bug with `not_equal_scalar` function always returning the wrong result.

### ISSUE THREAD

**Issue by zoj613** [owlbarn/owl#674]
There is a serious bug with the implementation of `not_equal_scalar` function. In it's current state, it is an alias for `equal_scalar` since it returns the opposite of it's expected result. Here is a reproducing example:

```ocaml
module Ndarray = Owl.Dense.Ndarray.Generic;;
let x = Ndarray.create Bigarray.Int64 [|2;2;2|] 10L;;
Ndarray.equal_scalar x 10L;;
(*- : bool = true *)
Ndarray.not_equal_scalar x 10L;;
(*- : bool = true *)
```
Evidently, `not_equal_scalar` is supposed to return `false`. With the help of my LSP server, I was able to jump to the definition of this function and to my surprised found that the underlying implementation of both functions is identical. See: 
https://github.com/owlbarn/owl/blob/455bde1ac6f1f94c24eca43bdbe45af4f80e139f/src/owl/dense/owl_dense_ndarray_generic.ml#L1940-L1942

Basically the implementation of `equal_scalar` was copied over to `not_equal_scalar`. I looked to see if there is an implementation function named `_not_equal_scalar` and it turns out it does exist at: https://github.com/owlbarn/owl/blob/455bde1ac6f1f94c24eca43bdbe45af4f80e139f/src/owl/core/owl_ndarray_maths.ml#L1618

So a simple fix for this bug is to just replace 
```ocaml
let not_equal_scalar x a = _owl_equal_scalar (kind x) (numel x) x a = 1 
```
with
```ocaml
let not_equal_scalar x a = _owl_not_equal_scalar (kind x) (numel x) x a = 1 
```
Cc @mseri @jzstark
